src/hud/widgets_hud.rs

```diff
@@ -14,23 +13,22 @@ pub struct WidgetsHud {
     texture_bind_group: BindGroup,
     geometry_buffers: GeometryBuffers<u16>,
     pub hotbar_cursor_position: usize,
+
+    ui_scale_x: f32,
+    ui_scale_y: f32,
 }
 
 impl WidgetsHud {
     pub fn new(render_context: &RenderContext) -> Self {
         let (_, texture_bind_group) = Self::create_textures(render_context);
-
-        let geometry = Geometry {
-            vertices: VERTICES.to_vec(),
-            indices: INDICES.to_vec(),
-        };
-        let geometry_buffers =
-            GeometryBuffers::from_geometry(render_context, &geometry, BufferUsage::COPY_DST);
+        let geometry_buffers = Self::generate_geometry_buffer(render_context, 0.0, 0.0);
 
         Self {
             texture_bind_group,
             geometry_buffers,
             hotbar_cursor_position: 0,
+            ui_scale_x: 0.0,
+            ui_scale_y: 0.0,
```

Comment:
Perhaps it'd be nice to default these to the `UI_SCALE_{X,Y}` constants? That'll make sure that `WidgetsHud` is in an operable state right after `new`'ing it

Reply:
Trying to think of a way to only need to set the aspect ratio instead of scale_x and scale_y

Reply:
That sounds even better

Reply:
scale_x is just used in so many places and that would be replaced by scale_y which wouldn't actually work if we want to support the toolbar shrinking when the window width is too small. Currently I check if the toolbar would be clipped and then shrink it. We could just ignore that case